### PR TITLE
Result.py: Add actions attribute

### PR DESCRIPTION
**Pull request by akshatkarani**
This adds a new attribute `actions` to the Result class. It is a list of action objects which are specific to the origin to that Result.
This also makes changes in `ConsoleInteraction` module and `Processing` module to incorporate these additions and support bears to define their own actions.

### CODE REVIEW

File: coalib/processes/Processing.py [hunk 1]
```diff
@@ -145,30 +150,34 @@ def autoapply_actions(results,
             else:
                 not_processed_results.append(result)
                 continue
+        if action not in bear_actions or action in result.actions:
```
Comment (akshatkarani) [inline coalib/processes/Processing.py]: This condition ensures that if action is in `bear_actions` then it must be in `result.actions`.

**Comment by abhishalya** [comment]
Ping @Makman2

**Comment by abhishalya** [comment]
Can you change the PR title = commit shortlog.

Its a bit confusing everytime I see it :P

**Comment by Makman2** [comment]
From my side nothing more to complain :) can be merged.

**Pull request merged**